kube-app-testing.sh

```diff
@@ -584,42 +533,8 @@ create_gs_cluster () {
     sleep 30
   done
 
-  info "Creating key-pair for tenant cluster access."
-  # create a key pair (must be stored directly in a variable)
-  _key_pair=$(curl ${GS_API_URL}/v4/clusters/${CLUSTER_ID}/key-pairs/ -X POST \
-    -H "Content-Type: application/json" \
-    -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" \
-    -d "$(gen_gs_blob keypair)")
-
-  # check that we actually got a key pair back
-  grep -q "certificate_authority_data" <<< $_key_pair
-  if [[ "$?" -gt 0 ]]; then
-    err "Key pair creation failed."
-    exit 3
-  fi
-
-  # parse required fields from key pair creation response in order to
-  # create a kubeconfig for the tenant cluster
-  TC_API_URI=$(curl -s -H "Authorization: giantswarm ${GSAPI_AUTH_TOKEN}" ${GS_API_URL}/v5/clusters/${CLUSTER_ID}/ | jq -r .api_endpoint)
-  CA_CERT=$(echo $_key_pair | jq -r '.certificate_authority_data | @base64')
-  CLIENT_CERT=$(echo $_key_pair | jq -r '.client_certificate_data | @base64')
-  CLIENT_KEY=$(echo $_key_pair | jq -r '.client_key_data | @base64')
-
-  # align config with kind clusters
-  if [[ ! -d ${CONFIG_DIR} ]]; then
-    mkdir ${CONFIG_DIR}
-  fi
-
-  info "Templating kubeconfig out to ${KUBECONFIG}"
-  # create the kubeconfig
-  gen_gs_blob kubeconfig ${KUBECONFIG}
-
-  # update Security Group to allow access
-  update_aws_sec_group ${CLUSTER_ID}
```

Comment:
this won't work - if you remove the call to the `update_aws_sec_group` function then none of the tests will work against GS clusters - `kubectl` in the CI job won't be able to talk to the GS cluster API.

Reply:
:bulb: so this is changing the security groups of the GS control plane and not the tenant cluster nodes?

Reply:
it's exposing port 443 for the tenant cluster k8s API externally - it doesn't affect the CP at all. It does 3 things:

1. finds the security group in the TC account tagged with the cluster ID
2. finds the rule in the security group which blocks access to 443
3. adds an ingress rule which opens up port 443 to 0.0.0.0/0 (if it doesn't already exist)

This is required because by default none of our tenant clusters are created with the k8s api exposed to the world, and it's not something you can specify when creating the cluster.

Reply:
So this is only required for running in CI without VPN access against a GS TC?

Reply:
that's correct, yeah. we investigated setting up vpn access etc but it wasn't really worth it - these clusters are short-lived so the danger of opening the k8s api up to the world is pretty minimal.

Reply:
Thanks for the clarification :)